python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py

```diff
@@ -263,17 +294,23 @@ async def on_messages_stream(
         result = await self._model_client.create(
             llm_messages, tools=self._tools + self._handoff_tools, cancellation_token=cancellation_token
         )
-
         # Add the response to the model context.
         self._model_context.append(AssistantMessage(content=result.content, source=self.name))
 
         # Run tool calls until the model produces a string response.
+        tool_call_iteration = 0
+        tool_call_msg: ToolCallMessage | None = None
+        tool_call_result_msg: ToolCallResultMessage | None = None
         while isinstance(result.content, list) and all(isinstance(item, FunctionCall) for item in result.content):
+            if tool_call_iteration >= self._max_tool_iterations:
```

Comment:
For cleaner code: check `max_tool_iterations` in constructor: it must be greater than or equal to 1. So you don't need to check this here.